Makefile

```diff
@@ -1,27 +1,96 @@
-FEATURES_INTERNALS=--features internals
+.DEFAULT_GOAL := help
+
+.PHONY: help
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+# -- variables --------------------------------------------------------------------------------------
+
+WARNINGS=RUSTDOCFLAGS="-D warnings"
+DEBUG_ASSERTIONS=RUSTFLAGS="-C debug-assertions"
 FEATURES_CONCURRENT_EXEC=--features concurrent,executable
 FEATURES_LOG_TREE=--features concurrent,executable,tracing-forest
 FEATURES_METAL_EXEC=--features concurrent,executable,metal
-PROFILE_OPTIMIZED=--profile optimized
-PROFILE_TEST=--profile test-release
 
-bench:
-	cargo bench $(PROFILE_OPTIMIZED)
+# -- linting --------------------------------------------------------------------------------------
+
+.PHONY: clippy
+clippy: ## Runs Clippy with configs
+	cargo +nightly clippy --workspace --all-targets --all-features -- -D warnings
+
+
+.PHONY: fix
+fix: ## Runs Fix with configs
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets --all-features
+
+
+.PHONY: format
+format: ## Runs Format using nightly toolchain
+	cargo +nightly fmt --all
+
+
+.PHONY: format-check
+format-check: ## Runs Format using nightly toolchain but only in check mode
+	cargo +nightly fmt --all --check
+
+
+.PHONY: lint
+lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, formatting)
+
+# --- testing -------------------------------------------------------------------------------------
+
+.PHONY: test
+test: ## Runs all tests
+	$(DEBUG_ASSERTIONS) cargo nextest run --cargo-profile test-release --features internals
+
+# --- docs ----------------------------------------------------------------------------------------
+
+.PHONY: doc
+doc: ## Generates & checks documentation
+	$(WARNINGS) cargo doc --all-features --keep-going --release
+
+.PHONY: mdbook
+mdbook: ## Generates mdbook
+	mdbook exec docs/
```

Comment:
Is `mdbook exec` a valid command? I thought it was `mdbook build` (i.e., based on [this](https://rust-lang.github.io/mdBook/cli/index.html)).

Also, I'd write the comment as "Generated mdBook documentation".